### ISSUE TITLE
`VkImageMemoryBarrier-image-00004` doesn't take timeline semaphores into account

### ISSUE THREAD

**Issue by saran-t** [KhronosGroup/Vulkan-ValidationLayers#4427]
**Describe the Issue**
I have a `VkImage` that is used as a color attachment output on the graphics queue family and as a transfer source on the dedicated transfer queue family. The command buffer on the graphics queue consists of a render pass followed by an image release barrier. The command buffer on the transfer queue consists of an image acquire barrier followed by a `vkCmdCopyImageToBuffer`.

The `vkQueueSubmit` to the graphics queue signals timeline semaphore (increments by one), and the `vkQueueSubmit` to the transfer queue waits on the same semaphore.

If I the transfer queue command buffer before the graphics queue command buffer, I get a `UNASSIGNED-VkImageMemoryBarrier-image-00004` validation error (full message below), however this call pattern seems to be valid since the timeline semaphore guarantees that the acquire barrier is only triggered after the graphics queue has executed its release barrier. (Empirically, over thousands of frames I do not detect any corrupted buffer that would indicate a data race.)

**Valid Usage ID**
```Validation Error: [ UNASSIGNED-VkImageMemoryBarrier-image-00004 ] Object 0: handle = 0x7863c682948, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x4acfa767 | vkQueueSubmit(): in submitted command buffer VkImageMemoryBarrier acquiring ownership of VkImage (VkImage 0x1c000000001c[]), from srcQueueFamilyIndex 0 to dstQueueFamilyIndex 1 has no matching release barrier queued for execution.```

**Environment:**
 - OS: Debian-based (kernel version 5.17.11)
 - GPU: Nvidia GeForce RTX 3090
 - SDK version: 1.3.204.1
 - Options enabled (synchronization, best practices, etc.): `VK_EXT_debug_utils`

**Additional context**
I'm a Vulkan novice so it's entirely possible that I'm in fact misinterpreting the spec.

**Comment by jeremyg-lunarg** [comment]
This is a duplicate of #2441. The validation for the above error occurs at submission time but it really should occur at execution time. Fixing this problem requires quite a bit of rearranging but we are slowly getting to the point where it can be fixed.

**Comment by artem-lunarg** [comment]
Fixed by https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/10316
